### ISSUE TITLE
solo.searchButton(String text, boolean onlyVisible) is not working properly.

### ISSUE THREAD

**Issue by renas** [RobotiumTech/robotium#178]
_From [anandpai...@gmail.com](https://code.google.com/u/116639303515044083042/) on April 07, 2011 22:33:16_

What steps will reproduce the problem? 1.Launch the contacts.
2.From menu-> manage contacts-> select an account which has no contacts
3.call the solo.searchButton(String text, boolean onlyVisible) api. What is the expected output? What do you see instead? Actual:

======
Returns true even though the button is not present.

Expected:

========
Should return false when button is not available. What version of the product are you using? On what operating system? Android2.3 and robotium 2.2 Please provide any additional information below.

_Original issue: http://code.google.com/p/robotium/issues/detail?id=99_

**Comment by renas** [comment]
_From [renasr...@gmail.com](https://code.google.com/u/108052454046732184536/) on April 08, 2011 00:42:19_

Are you searching for something that was seen in the previous screen?

**Comment by renas** [comment]
_From [anandpai...@gmail.com](https://code.google.com/u/116639303515044083042/) on April 08, 2011 03:23:33_

No i am searching in the current opened screen and have kept the delay to confirm that.

**Comment by renas** [comment]
_From [renasr...@gmail.com](https://code.google.com/u/108052454046732184536/) on April 08, 2011 05:00:20_

So you are searching for something in the current screen that is not there and you get true back? Are you sure it is not there? Have you tried onlyVisible = true?

**Comment by renas** [comment]
_From [anandpai...@gmail.com](https://code.google.com/u/116639303515044083042/) on April 08, 2011 06:48:57_

yes i have tried with the onlyVisible = true

**Comment by renas** [comment]
_From [renasr...@gmail.com](https://code.google.com/u/108052454046732184536/) on April 08, 2011 08:07:48_

The button was first available and then became unavailable?

**Comment by renas** [comment]
_From [anandpai...@gmail.com](https://code.google.com/u/116639303515044083042/) on April 08, 2011 08:57:15_

Yes the button was available and when deleted the contacts from listview and again launched manage contacts screen then the buttons are not available but was returning "true".

**Comment by renas** [comment]
_From [renasr...@gmail.com](https://code.google.com/u/108052454046732184536/) on April 10, 2011 00:41:43_

This is due to a bug in the Android platform. Sometimes old views are not cleaned up and there is no way to know that they are not showing. If you ask the view it self it will say that it is shown and it will behave like it is active even though it is not. Unfortunately there is nothing that can be done.

**Status:** WontFix

**Comment by renas** [comment]
_From [anandpai...@gmail.com](https://code.google.com/u/116639303515044083042/) on April 10, 2011 00:57:20_

can fix this problem by closing all the activities and again relaunching the activity.

**Comment by renas** [comment]
_From [renasr...@gmail.com](https://code.google.com/u/108052454046732184536/) on April 10, 2011 08:30:32_

Yes that should work. Anything that starts the garbage collector.

**Comment by renas** [comment]
_From [anandpai...@gmail.com](https://code.google.com/u/116639303515044083042/) on April 10, 2011 13:35:07_

How can i start the instrumented activity again in the same testcase as only once it can be launched with instruementation.

**Comment by renas** [comment]
_From [dadea...@gmail.com](https://code.google.com/u/110324432572714529946/) on April 13, 2011 07:48:03_

I am using 2.2 as well and I have asked it to find a fake button that should not be on there at all and it still returns a positive response. I have included the following code:
solo.searchButton("Help me Rhonda!",1, true);

There is no button with that string associated with it on the screen.  We are able to make solo.clickOnButton("Call") based on their string names.
